app/src/main/java/com/stevesoltys/seedvault/plugins/webdav/WebDavStoragePlugin.kt

```diff
@@ -0,0 +1,199 @@
+package com.stevesoltys.seedvault.plugins.webdav
+
+import android.content.Context
+import android.util.Log
+import at.bitfire.dav4jvm.DavCollection
+import at.bitfire.dav4jvm.Response.HrefRelation.SELF
+import at.bitfire.dav4jvm.exception.NotFoundException
+import at.bitfire.dav4jvm.property.DisplayName
+import at.bitfire.dav4jvm.property.ResourceType
+import com.stevesoltys.seedvault.plugins.EncryptedMetadata
+import com.stevesoltys.seedvault.plugins.StoragePlugin
+import com.stevesoltys.seedvault.plugins.chunkFolderRegex
+import com.stevesoltys.seedvault.plugins.saf.FILE_BACKUP_METADATA
+import com.stevesoltys.seedvault.plugins.saf.FILE_NO_MEDIA
+import com.stevesoltys.seedvault.plugins.tokenRegex
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import java.io.IOException
+import java.io.InputStream
+import java.io.OutputStream
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+
+internal class WebDavStoragePlugin(
+    context: Context,
+    webDavConfig: WebDavConfig,
+    root: String = DIRECTORY_ROOT,
+) : WebDavStorage(webDavConfig, root), StoragePlugin<WebDavConfig> {
+
+    override suspend fun test(): Boolean {
+        val location = baseUrl.toHttpUrl()
+        val davCollection = DavCollection(okHttpClient, location)
+
+        val webDavSupported = suspendCoroutine { cont ->
+            davCollection.options { davCapabilities, response ->
+                debugLog { "test() = $davCapabilities $response" }
+                if (davCapabilities.contains("1")) cont.resume(true)
```

Comment:
What do these checks do? Couldn't we do `containsAny` or similar?

Reply:
Got it thanks. I was just a bit confused as someone reading the code as to what this does, so I figure it could it be helpful to provide the above information in a comment or maybe make these constants to indicate what they are.